boost_manager/src/updater.rs

```diff
@@ -116,31 +119,49 @@ where
 
             // get a list of all the activations DB ids which form part of the batch
             let ids: Vec<u64> = batch.iter().map(|sp| sp.location).collect();
-
-            let solana_txn = self.solana.make_start_boost_transaction(batch).await?;
-            let transaction_id = solana_txn.get_signature().to_string();
-            // update the batch in the db with the txn id
-            db::save_batch_txn_id(&self.pool, &transaction_id, &ids).await?;
-
-            // if activations were processed successfully then
-            // update their status in the DB to success
-            // if not processed successfully then bump their retry count
-            // if retry count is below max retries then the activations
-            // will be retried next tick
-            match self.solana.submit_transaction(&solana_txn).await {
-                Ok(()) => {
-                    self.handle_submit_txn_success(&ids, batch_size, activations_count)
-                        .await?;
-                }
-                Err(e) => {
-                    tracing::warn!("submit txn failed, error: {}", e);
-                    self.handle_submit_txn_failure(&ids, batch_size).await?;
-                }
-            };
+            let txn = self.solana.make_start_boost_transaction(batch).await?;
+            let mut signed_txn = self.sign_and_prep_txn(&txn, &ids).await?;
+
+            // handle retries, if we encounter a blockhash not found error
+            // resign the txn with the latest blockhash before next retry attempt
+            let mut attempt: u32 = 1;
+            const MAX_ATTEMPTS: u32 = 10;
+            loop {
+                match self.solana.submit_transaction(&signed_txn).await {
+                    Ok(()) => {
+                        self.handle_submit_txn_success(&ids, batch_size, activations_count)
+                            .await?;
+                        break;
+                    }
+                    Err(err) if err.is_error_blockhash_not_found() && attempt < MAX_ATTEMPTS => {
+                        tracing::error!("block hash not found..possibly stale block hash, resigning txn and retrying");
```

Comment:
Is there any information we can add here about why the blockhash went stale?

Reply:
We look for a specific error which is blockhash not found.  This can only mean the blockhash submitted in the txn was incorrect from the first instance ( unlikely ) or it is now stale and it can only go stale if that blockhash is older than 150 blocks ( afaik )